processing/src/main/java/io/druid/query/search/SearchBinaryFn.java

```diff
@@ -62,24 +63,49 @@ public SearchBinaryFn(
       return arg1;
     }
 
+    final int limit = gran instanceof AllGranularity ? this.limit : -1;
```

Comment:
This merging process removes duplicates and cannot apply limit in early stage. It was already implemented like that before this.